src/graphics/graphics_texture.js

```diff
@@ -358,6 +358,85 @@ pc.extend(pc, function () {
          */
         upload: function () {
             this._needsUpload = true;
+        },
+
+        getDDS: function () {
+            if (this.format!=pc.PIXELFORMAT_R8_G8_B8_A8) {
+                console.error("This format is not implemented yet");
+            }
+            var fsize = 128;
+            var i = 0;
+            var j;
+            var face;
+            while(this._levels[i]) {
+                if (!this.cubemap) {
+                    var mipSize = this._levels[i].length;
+                    if (!mipSize) {
+                        console.error("No byte array for mip " + i);
+                        return;
+                    }
+                    fsize += mipSize;
+                } else {
+                    for(face=0; face<6; face++) {
+                        var mipSize = this._levels[i][face].length;
+                        if (!mipSize) {
+                            console.error("No byte array for mip " + i + ", face " + face);
+                            return;
+                        }
+                        fsize += mipSize;
+                    }
+                }
+                var mipSize
+                fsize += this._levels[i].length;
+                i++;
+            }
+
+            var buff = new ArrayBuffer(fsize);
+            var header = new Uint32Array(buff, 0, 128 / 4);
+
```

Comment:
If we'll have to extend this for many different pixel formats, some things can be clarified. But for now it's enough, it's just the fastest way to write it.